test/specs/list_spec.txt

```diff
@@ -39,6 +42,9 @@ out("\n");
 </ul>
 ===
 
+^^^
+    if (var_size1) {
```

Comment:
this can't be right. there must be a loop

Reply:
There **is** a loop, but this condition checks whether it has to be entered at all:
```
    const var_size1 = lengthOf(var_collectionVar0);
->  if (var_size1) {
      out("<ul>");
      for (const var_index2 of Object.keys(var_collectionVar0)) {
        ...
      }
```
That's the reason I put the generated location on the `if` conditional: otherwise, if the collection has no entries, an eventual breakpoint wouldn't be triggered at all.

Reply:
but where is the rest of the code? I would have expected that the output of the inner item is also source-mapped

Reply:
The inner item **is** mapped, on line 47 of `list_spec.txt` below, the detailed mapping reads:
```
->  if (var_size1) {
      out("<ul>");
      for (const var_index2 of Object.keys(var_collectionVar0)) {
        const item = Array.isArray(var_collectionVar0) ? var_collectionVar0[var_index2] : var_index2;
        const itemList = listInfo(var_index2, var_size1);
        out("\n    <li>");
        const var_3 = item;
->      out(var_3);
        out("</li>\n");
      }
    }
```

Reply:
ah...ok. makes sense now....